src/native/credentials.c

```diff
@@ -73,6 +73,10 @@ struct aws_credentials *aws_credentials_new_from_java_credentials(JNIEnv *env, j
         aws_jni_byte_cursor_from_jbyteArray_release(env, session_token, session_token_cursor);
     }
 
+    (*env)->DeleteLocalRef(env, access_key_id);
```

Comment:
I would like to have one place to clean things up even if we error out earlier.

Reply:
Also, add a comment here
```suggestion
   /* When the local reference created by a thread from C, JVM will not be able to clean it up automatically. */
```

Reply:
According to https://docs.oracle.com/en/java/javase/11/docs/specs/jni/functions.html, yes null is alowed.

Reply:
I modified it to check for null, ensuring consistency, and added a block for clean-up. I didn't move the cursor clean-up as I believe it would make the code messier.